_posts/2021-04-21-frequency-band-control-in-unetstack.md

```diff
@@ -0,0 +1,199 @@
+
+### The need to control centre frequency and bandwidth 
+
+Acoustic modems transmit physical sound waves via a transducer, typically a piezo electric device. Such sound emitters have an ideal resonance frequency F and a Q factor of the order of 0.3. This means that the efficient region of the frequencies it can transmit centres around the resonance frequency with a bandwidth of about 0.3 x F. Q may be higher or lower depending on the exact transducer.
+
+The transducer in medium range Subnero modem has a transducer with a resonance close to 26kHz << check>>. The recommended default bandwidth is B=12kHz which is half of the computationally possible bandwidth of 24kHz. Thus an effective transmit modulation scheme can be set such that the centre `C=26kHz` and `B=12kHz`.
+
+Another reason may be due to acoustic underwater channel characteristics. A most basic property is that higher frequencies get absorbed more and in general have lower transmission range. For this or other reasons related to the channel, the modem user may want to customize the frequencies transmitted by the modem. 
+
+Multiple Access is another case where users may want to control the band used in each modem. One common multiple access method is Frequency Domain Multiple Access (FDMA). In this scheme, different modems or modem pairs may be allocated a different frequency band for transmission to avoid collision. Simultaneous transmission-reception by two or more different and colocated modem pairs are possible using FDMA. 
+
+Sometimes, there may be a frequency band the user may want to exclude from usage, due to it being occupied by other sources, noise etc. 
+
+## Understanding baseband signal generation, transmission 
+
+It's good to have a basic understanding of how baseband signals are created and their frequency content. We can try out how it works in Subnero Modems using some examples.
+
+We shall use Binary Phase Shift Keying (BPSK) for this exercise.
+
+Create a baseband complex signal as follows. A complex baseband signal point [1,0] refers to the real and imaginary components of one baseband sample. So in the following signal, all imaginary values are 0.
+```
+> s = [1,0]*12000
+[1, 0, 1, 0, 1, 0, 1, 0, 1, 0, 1 <<snip>> 0, 1, 0, 1, 0, 1, 0, 1, 0, 1, 0]
+
+```
+
+Transmit this 1s long signal (12000 samples at a sample rate of 12kHz).
+
+```
+> phy << new TxBasebandSignalReq(signal: s)
+AGREE
+phy >> TxFrameStartNtf:INFORM[txTime:100610682 txDuration:1000416]
+phy >> TxFrameNtf:INFORM[txTime:100593099]
+```
+You should hear a tone sent out by the Unet audio running in the laptop.
+
+Passband PSD plot from the diagnostics scope shows this carrier only signal at nearly 12kHz.
+
+(settings used: plvl=-25, plot trigger set to -60, you can experiment with power and trigger to get a snapshot depending on your environment)
```

Comment:
Rather than give settings like this, would be nice to walk the user step-by-step through the process, including any necessary settings

Reply:
This is the main task remaining for me to fix.